provider/datagen/src/driver.rs

```diff
@@ -334,74 +354,110 @@ impl DatagenDriver {
+                                        return Ok(());
+                                    } else {
+                                        // Not a match: we must include this
+                                        break;
+                                    }
                                 }
                             }
-                        }
-                        // Did not find a match: export this payload
-                        sink.put_payload(key, locale, payload).map_err(|e| {
-                            e.with_req(
-                                key,
-                                DataRequest {
-                                    locale,
-                                    metadata: Default::default(),
-                                },
-                            )
-                        })
-                    })?
-                }
-                FallbackMode::Hybrid | FallbackMode::Preresolved => {
-                    locales_to_export.into_par_iter().try_for_each(|locale| {
-                        if let Some(payload) = load_with_fallback(key, &locale) {
-                            sink.put_payload(key, &locale, &payload?)
-                        } else {
-                            Ok(())
-                        }
-                        .map_err(|e| {
-                            e.with_req(
-                                key,
-                                DataRequest {
-                                    locale: &locale,
-                                    metadata: Default::default(),
-                                },
-                            )
+                            // Did not find a match: export this payload
+                            sink.put_payload(key, locale, payload).map_err(|e| {
+                                e.with_req(
+                                    key,
+                                    DataRequest {
+                                        locale,
+                                        metadata: Default::default(),
+                                    },
+                                )
+                            })
+                        })?;
+
+                    // Slowest locale calculation:
+                    payloads
+                        .iter()
+                        .map(|(locale, (_payload, duration))| {
+                            (*duration, locale.write_to_string().into_owned())
                         })
-                    })?
+                        .max()
                 }
+                FallbackMode::Hybrid | FallbackMode::Preresolved => locales_to_export
+                    .into_par_iter()
+                    .map(|locale| {
+                        let instant2 = Instant::now();
+                        load_with_fallback(key, &locale)
+                            .transpose()?
+                            .map(|payload| sink.put_payload(key, &locale, &payload))
```

Comment:
nit: I'm not convinced the functional style is clearer here, especially with the timing code, where you kinda actually do need to know where it should run. I think we should revert to the older imperative style here and just add timing

Reply:
In general I agree but in this case I was confused why we had an expression like

```rust
if let Some(payload) = load_with_fallback(key, &locale) {
    sink.put_payload(key, &locale, &payload?)
} else {
    Ok(())
}
// followed by other chaining logic
```

Also note that the majority of the diff in this branch is due to changing `try_for_each` to `map` so that I can call `.max()` on the result.

I will do another round of making this more clear though. I realized that I might be losing the error context with the new `.transpose()?`